packages/cli/src/metadataGeneration/typeResolver.ts

```diff
@@ -632,87 +660,376 @@ export class TypeResolver {
+      name = this.context[name].name; //Not needed to check unicity, because generic parameters are checked previously
+    } else {
+      const declarations = this.getModelTypeDeclarations(type);
+
+      //Two possible solutions for recognizing different types:
+      // - Add declaration positions into type names (In an order).
+      //    - It accepts multiple types with same name, if the code compiles, there would be no conflicts in the type names
+      //    - Clear namespaces from type names.
+      //    - Horrible changes can be in the routes.ts in case of teamwork,
+      //        because source files have paths in the computer where data generation runs.
+      // - Use fully namespaced names
+      //    - Conflicts can be recognized because of the declarations
+      //
+      // The second was implemented, it not changes the usual type name formats.
+
+      const oneDeclaration = declarations[0]; //Every declarations should be in the same namespace hierarchy
+      const identifiers = name.split('.');
+      if (ts.isEnumMember(oneDeclaration)) {
+        name = identifiers.slice(identifiers.length - 2).join('.');
+      } else {
+        name = identifiers.slice(identifiers.length - 1).join('.');
+      }
 
-    // Can't invoke getText on Synthetic Nodes
-    let resolvableName = node.pos !== -1 ? node.getText() : (type as ts.Identifier).text;
-    if (node.pos === -1 && 'typeArguments' in node && Array.isArray(node.typeArguments)) {
-      // Add typearguments for Synthetic nodes (e.g. Record<> in TestClassModel.indexedResponse)
-      const argumentsString = node.typeArguments.map(arg => {
-        if (ts.isLiteralTypeNode(arg)) {
-          return `'${String(this.getLiteralValue(arg))}'`;
+      let actNode = oneDeclaration.parent;
+      let isFirst = true;
+      while (!ts.isSourceFile(actNode)) {
+        if (!(isFirst && ts.isEnumDeclaration(actNode)) && !ts.isModuleBlock(actNode)) {
+          if (ts.isModuleDeclaration(actNode)) {
+            const moduleName = actNode.name.text;
+            name = `${moduleName}.${name}`;
+          } else {
+            throw new GenerateMetadataError(`This node kind is unknown: ${actNode.kind}`, type);
+          }
         }
-        const resolved = this.attemptToResolveKindToPrimitive(arg.kind);
-        if (resolved.foundMatch === false) return 'any';
-        return resolved.resolvedType;
-      });
-      resolvableName += `<${argumentsString.join(', ')}>`;
+        isFirst = false;
+        actNode = actNode.parent;
+      }
+
+      const declarationPositions = declarations.map(declaration => ({
+        fileName: declaration.getSourceFile().fileName,
+        pos: declaration.pos,
+      }));
+      this.current.CheckModelUnicity(name, declarationPositions);
     }
+    return name;
+  }
 
-    const name = this.contextualizedName(resolvableName);
+  private calcMemberJsDocProperties(arg: ts.PropertySignature): string {
+    const def = TypeResolver.getDefault(arg);
```

Comment:
If two types differ only in jsdoc, then their name needs to differ, too. Because of it, I need to inject the jsdoc parameters into the type name.